pkg/azure/api/api.go

```diff
@@ -386,6 +410,34 @@ func (c *Client) GetInstances(ctx context.Context, subnets ipamTypes.SubnetMap)
 	return instances, nil
 }
 
+var vmssRe = regexp.MustCompile(`[Vv]irtual[Mm]achine[Ss]cale[Ss]ets\/(.*)\/[Vv]irtual[Mm]achines\/(.*)`)
```

Comment:
nit: this matches the VMSS instance ID (not the VMSS itself), so consider calling it `vmssInstanceRe`?

Reply:
also why `(.*)` which could match "" instead of `(.+)` to match only non-empty string?

Reply:
Removed the regexp, see https://github.com/cilium/cilium/pull/37430#discussion_r1943068320